proposals/exp-show-target.md

```diff
@@ -0,0 +1,214 @@
+* Parameters files may include values that aren't tracked by DVC at all (for
+  example, debug level).
+* The parameters may be used in a different pipeline branch (for example, the
+  `topic_modeling` stage is in a separate branch of the pipeline from the
+  `evaluate` stage, which produces the relevant experiment metrics).
+* The parameters may be used in a downstream pipeline branch (for example, the
+  post-processing `combine` stage parameters aren't relevant to experiment
+  evaluation).
+
+```console
+                +-------------------+
+                | data/data.xml.dvc |
+                +-------------------+
+                          *
+                          *
+                          *
+                     +---------+
+                     | prepare |
+                     +---------+
+                          *
+                          *
+                          *
+                    +-----------+
+                    | featurize |
+                  **+-----------+**
+              ****       *         ****
+          ****           *             ***
+        **              *                 ***
++-------+               *                    **
+| train |             **                      *
++-------+            *                        *
+         **        **                         *
+           **    **                           *
+             *  *                             *
+        +----------+                  +----------------+
+        | evaluate |                  | topic_modeling |
+        +----------+                  +----------------+
+                   ***             ****
+                      *         ***
+                       **     **
+                     +---------+
+                     | combine |
+                     +---------+
+```
+
+There are no apparent reasons that users want to see untracked parameters in
+`dvc exp show`. Unlike `dvc params diff` and `dvc metrics diff`, `dvc exp show`
+is not applicable to arbitrary files. Likewise, there are no apparent reasons
+that users want to see parameters or metrics from irrelevant stages of their
+pipelines. Users could keep untracked parameters in other files if they don't
+want them shown in `dvc exp show`, and they could exclude parameters or metrics
+from irrelevant stages, but DVC already has enough information to understand
+that these should not be shown.
+
+# Detailed design
+
+`dvc exp show` could take stages as (positional?) arguments, similar to `dvc exp
+run`. The help message could be similar:
+
+```console
+positional arguments:
+stages               Stages for which to print experiment info. 'dvc.yaml'
+by default.
+```
+
+If no stages are passed, parameters and metrics defined in any stage would be
+shown. This differs from current behavior only because untracked parameters
+would be excluded.
+
+If stages are passed, parameters and metrics defined anywhere in the pipeline
+for those stages would be shown.
```

Comment:
Why not just for those specific stages? Wouldn't mind incorporating the `-[-up/down]stream` options to this proposal TBH. Full makeover 💇 💅

Reply:
Or `--single-item` as mentioned a bit further down.

Reply:
> this is out of scope for now

The proposal format seems a bit long anyway, so why not cover as much as possible? Reuse the motivation

Reply:
> The proposal format seems a bit long anyway, so why not cover as much as possible? Reuse the motivation

True, I'm used to trying to limit scope for issues, but one advantage of this format is to holistically discuss related issues in one cohesive doc. Thoughts on `--single[-item]`?

Reply:
> Thoughts on --single[-item]?

If the idea here is to remove noise from the very busy `exp show` table maybe single-item should be the default and have `--upstream` instead (or some other names). No strong opinion though.

Would also throw in `--downstream` for consistency in either case.

Reply:
Again, my initial reaction is to follow the `repro` conventions, which would do upstream by default, but also include:

* `-s, --single-item`
* `-p, --pipeline`
* `--downstream`

Reply:
No strong opinion